### DISCUSSION TITLE
FeaCoupling of an External floater (platform+mooring) with Openfast #791

### DISCUSSION THREAD

**Discussion by ZahidUllah33400** [OpenFAST/openfast#801] [Q&A]
Hello
I have been using Fast v8 and currently, I have started learning Openfast. I am interested in coupling of an external floater with the Openfast. what I want to do is that I want to develop my code that simulates the floater motion (floater: mooring lines+platform) and I want to use those six degrees of freedom motion of floater as an input argument for Openfast. This means I want to suppress the mooring module and platform kinematics of Openfast. Now I understand that the Elastodynamic module of Openfast includes the platform+tower+rotor+nacelle. Does this mean I have to somehow suppress the platform of the elastodynamic modules? As a beginner, I have few points in mind and would highly appreciate the expert's opinion on those.

My Goal:
At each time-step of the simulation, my code will provide the platform motion to the Openfast (at the base of the tower) and in return the Openfast would give me the reaction forces at the tower.

Key points of clarification:

To achieve my target I need to learn the development side of Openfast but I want to know where should I start.
Is it a difficult job and possible to be done in 4 months.
Can anyone gives me key step I have to follow to get to my target (just mention main things I have to learn to get to my target)
I will highly appreciate the expert's remarks and suggestions. Thanks

**Comment by jjonkman** [comment]
Dear Zahid,

Can you clarify what type of coupling you are envisioning?  Are you envisioning a one-way coupling where you first run your external floater code, from which motions of the floater would be calculated, then you would input these motions to OpenFAST in a subsequent simulation?  Or you are envisioning a two-way coupling where your external floater code is coupled to OpenFAST time step by time step such that the reaction loads at the tower impact the floater motion from your external floater code?  The former method is much easier to implement, but less accurate, than the second method.

The biggest challenge overall is that the ElastoDyn module of OpenFAST considers the motion of the floating platform as internal degrees of freedom, so, ElastoDyn wants an external module to provide load input of the platform (from hydrodynamics and moorings) rather than motion input.

Best regards,

**Comment by ZahidUllah33400** [comment]
Dear Dr. Jonkman

You have always been helpful and I was waiting for your valuable suggestions on the current issue. I am envisioning a two-way coupling where there is an exchange of parameters (forces and motion) at each time-step of simulation. I was also a bit concerned about the ElastoDyn module as it encompasses floating platform, tower, rotor, and nacelle. Is it possible to somehow suppress the platform part of the ElastoDyn module while the remaining tower+rotor+nacelle remains active?

**Comment by jjonkman** [comment]
Dear @ZahidUllah33400,

What you are proposing would require a rearrangement of the ElastoDyn equations of motion.  That is, without modification, the loads applied to the floating platform (apart from those transferred through the tower from the turbine) are known inputs to ElastoDyn and the platform accelerations are unknowns (solved via generalized F=ma), from which the velocities and displacements can be found via time-integration.  What you are proposing would swap this so that the acceleration is input and the loads are unknowns (solved via generalized F=ma).  This would not be an easy change to make.

Can you change your external floater code to calculate loads instead of motions?

Can you back up and clarify why you are trying to couple an external floater code with OpenFAST?  What functionality does your external floater code have that is not captured by OpenFAST already?

Best regards,

**Comment by ZahidUllah33400** [comment]
Dear Dr. Jonkman

Your description is helpful that without modification the known inputs are load rather than platform acceleration. In one of the recently published articles, the author has described coupling between Fast v7 and Ansys AQWA, and illustrations in the paper show accelerations as inputs. I want to do a similar coupling but with a different floater code.  I would highly appreciate your comments on that coupling procedure and if that (procedure) can be helpful. (article title: Development and Application of an Aero-Hydro-Servo-Elastic CouplingvFramework for Analysis of Floating Offshore Wind Turbines, https://doi.org/10.1016/j.renene.2020.07.134)
I will highly appreciate it if you can take a look at this article and give me your views on the coupling procedure. 

Regarding the functionality, the subroutine (for the platform) we want to couple with Openfast considers a flexible platform and will compute the hydrodynamic loading on it.

**Comment by jjonkman** [comment]
Dear @ZahidUllah33400,

Well, I looked briefly at this article and it does not mention that the FAST v7 source code was modified to accept platform acceleration inputs.  However, the paper links to a github site where the source code can be found.  I have not reviewed the files here, but you could review them and reach out to the authors of the paper for clarification.  The ElastoDyn module of OpenFAST was derived from the structural model of FAST v7, and both have the same formulation for load input and motion output of the platform.  So, I would guess that they rearranged the equations of motion for the platform DOFs as I summarized in my prior post.

FYI: The ability to model floating substructures as structural flexible bodies with hydro-elastic effects and the ability to calculate internal member-level loads was introduced in OpenFAST v2.6 and newer.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkam
I looked into the source code and the author has made some changes in the source code. The author changed the source code of Solver in FAST.f90 to assign values to platform displacements, velocities, and accelerations that are obtained from AQWA and transferred by the DLL.

As you mentioned the ability to model floating substructures as structural flexible bodies have been introduced in OpenFast v 2.6. I am interested in studying the modeling procedure (theory manual). I will appreciate it if you can refer me to any helpful document which explains the flexible platform modeling procedures in Openfast. I have looked into "openfast-readthedocs-io-en-main".

**Comment by jjonkman** [comment]
Dear @ZahidUllah33400,

The ability to model floating substructure flexibility and member-level loads and associated updates to SubDyn, HydroDyn, and the OpenFAST glue code has now concluded, including code development and verification against OrcaFlex for the TetraSpar in collaboration with Stiesdal.  The source code from https://github.com/OpenFAST/openfast/pull/537 has been reviewed and approved and released in OpenFAST v2.6 and newer.  However, the full-system models we’ve tested contain proprietary information, so, these models cannot be shared publicly; we’ve posted the need for an example FOWT model with substructure flexibility in this r-test issue: https://github.com/OpenFAST/r-test/issues/39.

The user-documentation has been updated, but is still incomplete.  The SubDyn and HydroDyn input files have changed and these changes are reflected in the updated r-test (https://github.com/OpenFAST/r-test).  There is some documentation of the input file changes in the OpenFAST readthedocs (https://openfast.readthedocs.io/en/main/source/user/api_change.html).  Likewise, the SubDyn documentation has been completely revamped, as included in the OpenFAST readthedocs (https://openfast.readthedocs.io/en/main/source/user/subdyn/index.html).  The HydroDyn outputs have been updated a bit, as documented in the updated OutListParameter.xlsx spreadsheet—see: https://openfast.readthedocs.io/en/main/_downloads/3f19498a5dc774461e022b671ff01ec6/OutListParameters.xlsx.  The theory basis on many of the new features is summarized in the following NREL technical report: https://www.nrel.gov/docs/fy20osti/76822.pdf.

Here are a few general guidelines for running OpenFAST with this new capability:
- Currently, modeling a system with floating platform flexibility requires one correction step in the OpenFAST glue code (`NumCrctn` = 1). We are not sure why this is yet (likely some reordering of the glue code will solve this problem); but hopefully we an solve this in a future release.
- The substructure is considered floating in SubDyn if there are no base reaction joints. In a floating system, the 6 Guyan modes become rigid-body modes; the elasticity of the substructure is represented by the Craig-Bampton and static modes.
- In SubDyn, we recommend setting `SttcSolve` = TRUE and `GuyanLoadCorrection` = TRUE for floating systems.
- Setting `GuyanLoadCorrection` = True ensures that the applied loads (gravity and buoyancy) are applied at the displaced orientation of the substructure (to get the correct hydrostatic and gravity restoring). While the elasticity of the substructure is otherwise considered fully, the impact of the displacement-based terms for the applied loads (gravity and buoyancy) are assumed dominated by the rigid-body motions (elasticity is neglected in the gravity and buoyancy applied loads).  This simplification was necessary to maintain linearity assumptions within SubDyn. The tower is likely the most flexible element of the support structure; as such, we recommend modeling the tower in ElastoDyn instead of SubDyn.
- To calculate the tower mode shapes for ElastoDyn, it is likely a reasonable assumption to assume the substructure is a 6-DOF rigid body.
- We recommend setting `IntMethod` = 3 in SubDyn for the best numerical stability.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman
The information you provided was very helpful and the reference documents helped me a lot in understanding and I am highly thankful for that. I do want your opinion on if I want to rearrange the ElastoDyn module such that it can accept motion as input and give me loads as output, Do I have to completely rearrange the equation of motion? I want to know I have to only make modifications in the ElastoDyn module or other modules will be subjected to modifications. Would it be possible to do it in 4 months? (I have attached a simple sketch showing what I want to do). I will highly appreciate your valuable suggestions.
![image](https://user-images.githubusercontent.com/77382406/131659721-1ed77d7e-d0f0-43a4-91b6-e5ba66b7cf7d.png)
t to do)

**Comment by jjonkman** [comment]
Dear @ZahidUllah33400,

Your sketch seems reasonable, but please note that it is missing the velocities (V) and accelerations (A) that must be passed into OpenFAST.  The implementation can likely be completed in 4 months, but I have not done it myself and can't guide you through all of the details.

You'll have to modify the source code of both ElastoDyn and the OpenFAST glue code.  You'll need to change ElastoDyn so that the 6-DOF motions of the platform (U,V,A) are passed as input instead of the 6-DOF platform loads, the 6-DOF platform loads (RF) are output instead of the 6-DOF motion of the platform, and rearrange the equations of motion so that the 6 platform loads are the unknowns instead of the 6 platform accelerations.  You'll need to change the OpenFAST glue code so that it calls your hydrodynamic subroutine and that the updated ElastoDyn inputs and outputs are transferred through this call.  I'm not sure if the coupling approach will be implicit or not (i.e. if the acceleration input to OpenFAST (A) is directly dependent on the reaction force (RF) output from OpenFAST).  If the coupling is implicit, you'll have to address that too (similar to how we solve implicit coupling between modules now in the OpenFAST glue code through a rigorous nonlinear input-output solve), otherwise the coupling between OpenFAST and your code would be numerically unstable without a very small time step.

I hope that helps.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

I am studying the related available materials to accurately complete this work. To successfully do this I need thoroughly study and understand the Equation of Motion of the ElastoDyn module. I searched many published reports but could not find any such published material which provides an in-detail explanation of the theoretical background (Equation of Motion) of the ElastoDyn module. I would appreciate it if you can refer me to any such published document.

Regards
Zahid

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

The theory basis of ElastoDyn is documented in what is referred to as the "Unofficial Theory Manual" on our forum (e.g., see: https://wind.nrel.gov/forum/wind/viewtopic.php?f=4&t=1532).  E-mail me and I'll forward you the documents.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr, Jonkman 
Thank you for the reference. I have received the unofficial theory manual.
Regards,
Zahid

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

As previously I shared with you (in the same post) an article that described the coupling of FAST v7 and AQWA (F2A). A recent article on the same coupling shows that the author did not rearrange the equation of motion but rather modify the solver subroutine of the FAST v7. I have highlighted the sentences which explain the modfication made in the solver subroutine. I would appreciate if you can give to highlighted sentences in section 3.3 of the attached article. I have also attached a snapshot of the changes in the solver subroutine (which shows how he impose the kinematics from AQWA in the solver subroutine).

Now coming to OpenFAST, one approach to as you have explained is to rearrange the equation of motion to make it such it can accept the platform motions as input rather than loads. I wanted to know if this approach (explain in the article) can also be adopted/used to make OpenFAST such it uses platform kinematics (U, V, A) obtained from another source (e.g., AQWA or Abaqus) during the solution.
your valuable comments help me a lot in learning and are always highly appreciated. 
![Uploading changes in solver subroutine of fastv7.JPG…]

[yang2021.pdf](https://github.com/OpenFAST/openfast/files/7237605/yang2021.pdf)

**Reply by ZahidUllah33400** [reply]
![changes in solver subroutine of fastv7](https://user-images.githubusercontent.com/77382406/134942215-c499ce8c-4ad9-4dc9-b2bb-e2be5b638746.JPG)

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

I did not review this paper originally, but I'm concerned that the approach implemented by Yang et al is not correct.  The equations of motion are coupled between the various DOFs.  So, simply overwriting the platform accelerations calculated internally within FAST by those computed externally by AQWA will result in an inconsistency in the equations of motion that is not resolved, i.e. the load output passed to AQWA from FAST will not be consistent with the acceleration input from AQWA to FAST.  To resolve this inconsistency, it would be necessary to rearrange the equations of motion as I outlined in my prior post.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman 
I was concerned about the approach used in that paper and therefore needed your viewpoint.  At the moment, I decided to change my floater code to calculate loads (as you suggested in one of the comments), and those loads will be transferred as Input to ElastoDyn and the equation of solution of the equation of motion will provide motions as output. The two-way transfer of motion and loads will take place at each time step of the simulation. I view it as using my external floater code instead of SubDyn. Apart from the floater code, I have to develop a DLL which will exchange the parameters between the OpenFAST and the external floater code. I think this way I don't have to make any changes in the OPenFAST source code. I would highly appreciate your valuable suggestions regarding this approach.

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

Sounds reasonable, but this may require many changes to your floater code?

The interface between ElastoDyn and SubDyn is not trivial, involving a sophisticated input-output solve to ensure the motion inputs to SubDyn and load inputs to ElastoDyn are internally consistent every time step (what we refer to as an "Option 1" solve in the OpenFAST glue code).  If your floater code functions the same was as SubDyn, then this scheme may work straight out, but again, this likely requires many changes to your floater code.  I would recommend reviewing the inputs and outputs of SubDyn and how they are used within the OpenFAST glue code so that you can mimic the same in your floater code.

Best regards,

**Reply by ZahidUllah33400** [reply]
Thank you Dr. Jonkman, I am thoroughly studying the input and output of SubDyn.

**Reply by ZahidUllah33400** [reply]
Dear Dr, Jonkman

I have been looking into the SubDyn module and also studied the theory manual of the SubDyn module. In SubDyn module floating platform is modelled using a beam element, and its equation of motion is solved to obtain the platform responses. I have one small query regarding this. When SubDyn module is enabled, then Platform is modelled using the beam element, in that case, I was curious about the 6 DoFs of the platform that are considered in the ElastoDyn module (among the 22 or 24 DoFs equation of motion, there are 6 platform DoF). Are those 6 DoFs somehow disabled as now the platform is modelled in SubDyn? 
Regards,
Zahid

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

The rigid-body motion of the floater is enabled through the 6 platform DoFs in ElastoDyn.  When modeling a floating platform with SubDyn enabled (`NReact` = 0), the Guyan modes in SubDyn characterize the rigid-body mass, center of mass, and inertia of the floater, but the DOFs are still solved in ElastoDyn (the corresponding mass and inertia terms in ElastoDyn can be zeroed because the mass/inertia are accounted for in the Guyan modes of SubDyn).  When modeling a floating platform with SubDyn enabled, the Craig-Bampton modes and static modes solved in SubDyn characterize the elasticity of the floater, which is added to the rigid-body motion.

Best regards

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

We are working on modeling the floater in Abaqus and the Abaqus and then coupling it with OPenFAST (in place of SubDyn). As you mentioned that the SubDyn module provides the Elastic Deformation and ElastoDyn provides the 6 DoF and then both are summed together. In our case both Elastic deformation+6 Motion are obtained from the Abaqus. Now for coupling do I have to somehow disable 6 DoF in ElastoDyn or is there any other way?

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

Again, if you wish for the 6-DOF rigid-body motion of the floater to be solved in ABAQUS, you would need to modify the source code of the OpenFAST glue code and ElastoDyn so that the 6-DOF motion is passed to ElastoDyn rather than output from ElastoDyn.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

Thank you for your valuable input. The needed changes in ElastoDyn module would be like reducing the equation of motion (F=Ma) by 6 DoF i.e. reduce the equation of motion from 24 - 6 (platform Dof) = 18 DoF?

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

Yes, that is correct, but the 6 tower-base loads can now be considered unknowns that need to be solved in the process (rather than the 6 unknown accelerations), as I discussed earlier.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman
If correctly have understood, you are referring to the term "T" in the load vector (in rows 1 to 6). As discussed earlier, does this mean i have to rearrange the Equation of motion to solve for load vector rather than acceleration? or it is possible to simply reduce the equation of motion to total of 18 equations instead of 24 by removing the first 6 equations (corresponding to platform DoFs).

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

This is probably just a matter of semantics.  Yes, the equations of motion are reduced from 24 to 18 DOFs because 6 of the DOFs are now known inputs.  However, the associated forces, which were previously known, are now not known, so, these must be solved in the process.  Here is a high-level summary of the approach:

At a high level, ElastoDyn solves equations of motion in the generalized form, M*qdd = F, where M is the mass matrix (depending on displacements, q), qdd are the accelerations, and F is the force vector (depending on displacements, q, velocities,  qd, and applied loads).  Previously, for a given set of known states (q, qd) and applied loads, M and F would be known and the equations of motion would be solved for the unknown accelerations qdd.  After your change, the equations of motion must be rearranged because now a subset of qdd is known.  Assume a partition where accelerations "qdda" are the unknowns and the accelerations "qddb" are known, then at least a portion of Fb must be unknown.  The equations of motion can be rearranged from their original formulation:

[ Maa Mab; Mba Mbb ] * { qdda; qddb } = { Fa; Fb }

After rearrangement:

[ Maa 0; -Mba I ] * { qdda; Fb } = { Fa - Mab * qddb; Mbb * qddb }

where I is the identity matrix.  I.e., the unknown accelerations qdda can first be solved based on the known accelerations qddb via

Maa * qdda = Fa - Mab * qddb

Then, the unknown forces in Fb can be found from:

Fb = Mba * qdda + Mbb * qddb

There are, of course, many details missing from this overview, but hopefully the point is illustrated.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman
I am thankful for the detailed explanation.

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

Regarding the coupling, I am also interested in coupling the AeroDyn with an Abaqus subroutine. The tower and RNA assembly will be modeled in Abaqus and it will provide the blade position, velocity to AeroDyn. In return, I want the AeroDyn to calculate the blade forces and moment. These forces will be input to Abaqus at each time step of the simulation. To do this kind of coupling, can I use the standalone AeroDyn? your comments are quite helpful in understanding the coupling and I would appreciate your valuable suggestions on the Abaqus-AeroDyn coupling. 
Regards,
Zahid

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

I'm sure this is also possible, as AeroDyn has been coupled to other structural solvers in addition to those (ElastoDyn, BeamDyn) available in OpenFAST.  You shouldn't need to use the AeroDyn driver or the OpenFAST glue code for this, just the AeroDyn module itself.

AeroDyn is written in Fortran with module-level inputs and outputs that are derived Fortran types (spatial meshes along the blades and tower).  The first question to ask is if ABAQUS can call AeroDyn and pass the required inputs directly, or if you need to develop some wrapper around AeroDyn to call and transfer the input/output data.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

I have looked into the standalone AeroDyn module and the theAeroDyn driver provides the input and that information that are usually provided by the ElastoDyn module (blade positions and velocities) in case of coupling with Openfast. So, as you explained, I don't have to use the AeroDyn module because that information will be provided to the AeroDyn module by the Abaqus and inflow wind. Now in the standalone AeroDyn module which I have downloaded contains an .exe file (pre complied) and there is no source, can I use this standalone module (without the driver) and call this .exe file from Abaqus. Or it would be easier for me to call separate the AeroDyn module from the coupled version of OpenFast and use that for coupling with Abaqus.

Additionally, I also wanted your views on the ServoDyn module. Similar to the AeroDyn module, is it possible to use the ServoDyn module alone and coupled it with Openfast. Basically i want to couple AeroDyn and ServoDyn module with abaqus. I would highly appreciate your valuable suggestions.

Regards
Zahid

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

I presume you mean that you "don't have to use the AeroDyn _driver_ because that information will be provided..."; if, so, I agree with this statement.

I'm not sure where you downloaded the standalone AeroDyn driver from, but in OpenFAST, NREL does not provide precompiled binary executables of the standalone AeroDyn module (or of any standalone drivers).  That said, NREL provides the compiling scripts so that users can compile the standalone module-level executables they want.  The standalone executable contains the compiled form of both the driver and module-level source code.

But you shouldn't use the standalone executable when coupling to ABAQUS.  Instead, you'd want ABAQUS to call the AeroDyn module rather than its driver.  A wrapper source code is likely needed to transfer the data output from and input to ABAQUS into a form that the AeroDyn module supports, including the Fortran-based mesh data structures.  If ABAQUS can't call Fortran, you'll likely need another wrapper to interface to a form that ABAQUS can call.

Moreover, AeroDyn needs both structural inputs (motions of the blades, hub, nacelle, and tower) and wind inflow inputs (ambient wind), so, presumably you'd need to interface ABAQUS to both the AeroDyn and InflowWind modules.

Similarly, if you want to couple ABAQUS with ServoDyn, you'd need to develop the appropriate interface between the two.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

Your explanation is quite helpful. I searched "Aerodyne standalone" and downloaded it from somewhere (exactly dont remember it). It contains a .exe file in the bin folder. Can you kindly share an NREL link for compiling scripts of the AeroDyn module (standalone) which contain module-level source code. Similarly, I would also appreciate it if you can share a link for the standalone SeroDyn and Inflow wind modules. Thanks
Regards,
Zahid

**Reply by jjonkman** [reply]
Dear @ZahidUllah33400,

The driver and module-level source code and compiling scripts are provided in the OpenFAST repository: https://github.com/OpenFAST/openfast.  Based on past communication, I believe you are running Windows (as opposed to Mac or Linux), in which case, I recommend compiling with Intel Fortran in Visual Studio using the Visual Studio solutions provided.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

I downloaded all the files from the above link, which contains several folders including "modules", and "vs-build". I go through all the folders. In the folder "Modules", the source codes of all the modules are given. As I want to compile the "AeroDyn", "Inflow Wind", and "ServoDyn" module as a standalone and then couple it Abaqus, therefore I used visual studio (with Intel Fortran compiler) to compile each module separately.  To compile I follow the following step:

1. go to "OpenFast folder > vs-build folder > InflowWind > open "InflowWind_driver.sln" in visual studio and click build. As a result in the bin folder, a .exe file "InflowWind_driver_Win32" was created.  

I follow the same process for AeroDyn module but the compilation was not successful due to some reason. 

I want to know if the above process was the correct way of compilation and using the modules as a standalone?
Secondly, in the FAST8 (i.e., https://www.nrel.gov/wind/nwtc/fastv8.html), precompiled modules are available. Can i use these directly by downloading them and coupling them with Abaqus?
I will appreciate your valuable comments.

**Comment by jjonkman** [comment]
Dear @ZahidUllah33400,

Yes, the process you've used to compile standalone InflowWind is correct.  What error you are getting when you do the same for standalone AeroDyn?

You can use the precompiled standalone drivers from FAST v8 for running standalone simulations, but coupling to Abaqus will require source code development and compiling.  And you should note that InflowWind, AeroDyn, and ServoDyn have all been updated in OpenFAST relative to FAST v8, so, using the older versions will not provide the most updated functionality.

Best regards,

**Reply by ZahidUllah33400** [reply]
Dear Dr. Jonkman

The error was related to the platform toolset version, but now i have successfully compiled the AeroDyn module as well. I am working to compile the ServoDyn module and then couple it with Abaqus through a subroutine VDLoad. I highly appreciate your help. Thank you.
